src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertEquals;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.FindPatientPage;
+import org.openmrs.contrib.qaframework.page.RegistrationEditSectionPage;
+import org.openmrs.contrib.qaframework.page.RegistrationSummaryPage;
+
+public class EditDemographicsAndContactInfoSteps extends Steps {
+
+	private TestData.PatientInfo testPatient;
+	private RegistrationEditSectionPage registrationEditSectionPage;
+	private RegistrationSummaryPage registrationSummaryPage;
+
+	@Before(RunTest.HOOK.SELENIUM_PATIENT_DEMOGRAPHICS)
+	public void visitPatientDashboard() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+		findPatientPage = (FindPatientPage) homePage.goToFindPatientRecord().waitForPage();
```

Comment:
@sherrif10  I am thinking since that action requires population of patients records which may take a few seconds then we need to wait for the page. What do you think?

Reply:
@sherrif10  and @kdaud  do I do away with  waiting for the Page? 
I know the test will pass but will it be the best way to implement this given that the page has to populate information from the api/ database?

Reply:
What happens when you remove it ?

Reply:
The test still passes, though you have not answered my question. Was it a vague?

Reply:
Not really please. Wanted to know whether the test fails without waiting method :+1:

Reply:
Some pages take long to load especially those containing  many elements with drop down lists  and its always a good ideal to invoke waitForPage() function to give allowance for these elements to be rendered on the page before any operation is  performed on the page.